internal/common/metricstorage/memStorage.go

```diff
@@ -0,0 +1,47 @@
+package metricstorage
+
+type MemStorage struct {
+	counters map[string]int
+	gauges   map[string]float64
+}
+
+func NewMemStorage() Storage {
```

Comment:
Не стоит возвращать тип интерфейса из фабричного метода. Так как Go использует duck typing, инстанс созданный `NewMemStorage()` уже будет реализовывать интерфейс если в нем реализованы соответствующие методы. 
```suggestion
func NewMemStorage() *MemStorage {
```

Reply:
++